Update to functional components & declare prop types in Health care application

longmd opened the pull request:
## Description
There are ESlint warnings in multiple files within the Health care application that are related to prop types. Many of the warnings are due to prop types not being declared for components, while a few are due to the order in which props are declared. This PR addresses those warnings and ensures all prop types are declared as expected. In many cases, components have been converted from class-based to functional components. 

## Original issue(s)
department-of-veterans-affairs/va.gov-team#44698

## Testing done
- [x] Unit tests

## Acceptance criteria
- [ ] Prop Type-related linting warnings are extinguished

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

longmd committed:
Update to functional components & declare prop types

longmd committed:
Update to functional components & declare prop types

nathanielbuck committed:
Add ToE to the Allow List (#21746)

brdunfield committed:
[Check-in] Fix for pre-check-in undefined token requests (#21753)

simiadebowale committed:
commented out test checks causing e2e tests to fail (#21754)

* commented out test checks causing e2e tests to fail

* updated

jerekshoe committed:
HLR & NOD | Fix issue with infinite loop (#21750)

* Fixes issue with nil ratingIssueSubjectText issues not being filtered out

* Fixing issue in NOD as well

justin-trieu committed:
[department-of-veterans-affairs/va-virtual-agent#562] Reverts copy changes that are pending approval. (#21755)

Co-authored-by: Kha Huynh <kha.huynh@thoughtworks.com>

ryanshaw committed:
40139 fix appointment list link (#21745)

* Add role attribute to details link

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

* Changed detail button to link element and added e.preventDefault

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

* Removed unneeded changes

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

* Removed unneeded changes

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

khathuynh-tw committed:
[department-of-veterans-affairs/va-virtual-agent#491] Fix flaky test (#21761)

OptionSelect committed:
42013 - [CDP] - Debt letter download page updates due to VBMS endpoint being back up (#21766)

* Updated alert content per convo with Jill and Denise regarding status of downloads.

* Update alert headline to match.

timwright12 committed:
Adding font-display for teamsite font loading (#21767)

Dr-Pongo committed:
[cFSR] Adding feature toggle for Combined FSR experience #21768

lzapson commented on src/applications/hca/components/CustomReviewField.jsx:
```diff
@@ -11,4 +12,8 @@ const CustomReviewField = ({
 );
 
+CustomReviewField.propTypes = {
+  children: PropTypes.object,
```
@Matth-Ng just an observation, I was wondering about PropTypes.object for children. It definitely makes sense as it is an object and in this case used to get properties from the object. In react Proptypes documentation I see it set to PropTypes.element or PropTypes.node.

asg5704 committed:
updates prod config for oauth (#21757)

dsrr47 committed:
44228 flag removal (#21770)

* added caution text to accreditation modal

* added production flags

* linter fixes

* removed environment flag

Co-authored-by: David <dsrr74@gmail.com>

dsasser committed:
VAMC-9797: Update crisis line phone number. (#21697)

Co-authored-by: Daniel Sasser <daniel.sasser@agile6.com>

brianseek committed:
Cypress screenshots (#21749)

* Create screenshot command and added to pre-check-in happy path.

* Expanded PCI screenshots.

* Added day-of screenshots.

* Added import files for screenshots.

* Refined screenshot specs and added to readme.

* Expanding more accordions.

* Added dummy test for screenshot specs.

* Updates to readme.

holdenhinkle committed:
don't create a string from an array (#21771)

* don't create a string from an array

* update all dependency values to arrays. update this tomorrow after job runs

asg5704 committed:
Adds OAuth logout functionality (#21703)

* initial commit

* initial commit

* adds/updates unit tests

* adds user-mocks for authBroker; updates user-nav.cypress.js to account for authBroker

* updates render error for time

* updates render-error-container

* reverts utils/api in favor of refresh-bug issue

dsrr47 committed:
44228 flag removal bug (#21772)

* added caution text to accreditation modal

* added production flags

* linter fixes

* removed environment flag

* removed duplicate text

Co-authored-by: David <dsrr74@gmail.com>

Commit:
add vet360 info to vaos mock api (#21742)

asg5704 committed:
updates correct config (#21776)

jenniemc committed:
fix for practitioner length (#21760)

* fix for practitioner length

* fix reading null id

longmd committed:
Update unit tests

longmd committed:
Merge

ddzz committed:
Increase retry wait time in header/footer E2E workflow (#21778)

longmd committed:
Adjust CustomReviewField prop type declaration

dsrr47 committed:
removed duplicate text (#21782)

Co-authored-by: David <dsrr74@gmail.com>

longmd committed:
Remove POA upload feature flag from Caregivers application (#21752)

* Remove POA upload feature flag

* Remove reference to feature flag from README

jerekshoe committed:
Burials | Added missing parameters to validateDate function (#21779)

* Added missing parameters to validateDate function

* Updates calls to validateDate from other validators

asg5704 committed:
Fixes the refresh bug (#21756)

* fixes the refresh bug

* updates unit tests

FernandoBoza committed:
imported MEB name valid & hooked it to first, mid, last (#21783)

Co-authored-by: fer.boza <fernando.boza@accenturefederal.com>

longmd committed:
Remove unnecessary ESLint disabling

dsasser committed:
VACMS-9899: Correct format of revision date in find-forms components. (#21784)

* VACMS-9899: Correct format of revision date in find-forms components.

* VACMS-9899: Update find-forms unit tests revision date sorting assertions to use the updated format for expected results and sorting.
- Update sorting logic to use display format constant rather than hard-coded format string.

Co-authored-by: Daniel Sasser <daniel.sasser@agile6.com>

maxx1128 committed:
[VA-21108] Remove locator tab indexes (#21751)

* VA-21108: Remove unneeded tab-inde from locator Operation Status

* VA-21108: Remove unneeded tab-inde from Location Operation Status

nathanielbuck committed:
Update URLs and Clean-Up Variables (#21774)

nathanielbuck committed:
Fix Preferred Contact Method Mapping (#21787)

holdenhinkle committed:
Update product directory scan (#21788)

* manually trigger daily-dependency-scan run

* try using the product directory github bot to grab last_updated values

* 'crossProductDependencies' was accidentally deleted from array. add it back.

* wip

* wip

* compare current and just scanned dependency values

* set gha workflow to run on a schedule again

* remove backslash after cron schedule string

khathuynh-tw committed:
Va virtual agent 480 (#21790)

* [department-of-veterans-affairs/va-virtual-agent#480] Adds end of auth topic utterance tracking.

Co-authored-by: Kha Huynh <kha.huynh@thoughtworks.com>

* [department-of-veterans-affairs/va-virtual-agent#480] Store utterance tracking flag in sessionStorage

Co-Authored-By: justin-trieu <89413920+justin-trieu@users.noreply.github.com>

Co-authored-by: justin-trieu <justin.trieu@thoughtworks.com>
Co-authored-by: justin-trieu <89413920+justin-trieu@users.noreply.github.com>

Mottie committed:
Supplemental Claims | Add evidence pages (#21769)

* Add evidence type page

* Add VA records page

* Add Private records pages

* Fix unit tests

* Revert validation range check

* Design tweaks

longmd committed:
Remove address autofill feature flag (#21785)

justin-trieu committed:
[department-of-veterans-affairs/va-virtual-agent#562] Adds 988 information to disclaimer (#21794)

Co-authored-by: Kha Huynh <kha.huynh@thoughtworks.com>

Co-authored-by: Kha Huynh <kha.huynh@thoughtworks.com>

asg5704 committed:
adds defaultOAuthWebOpts (#21792)

nathanielbuck committed:
Add Heading to the Direct Deposit Page (#21734)

bkjohnson committed:
Upgrade component-library to 11.2.1 (#21795)

* Bump component-library version

* Change class selector to id-selector

This is needed since the date web component now has markup more
consistent with the other form controls

Commit:
Remove the "Update browserslist" build step (flaky, not needed) (#21801)

Mottie committed:
Supplemental Claims | Add supporting evidence upload & summary pages (#21800)

* Always show "save" button text

* Remove unused pages

* Clean up private evidence upload

* Add other uploads page

* Add evidence summary page

* Add test data

* Fix summary page logic
